sdcm/provision/aws/capacity_reservation.py

```diff
@@ -118,10 +118,12 @@ def _get_supported_availability_zones(ec2, instance_types: List[str], initial_az
     @staticmethod
     def is_capacity_reservation_enabled(params: dict) -> bool:
         """Returns True if capacity reservation is enabled."""
+        is_single_dc = str(params.get("n_db_nodes")).isdigit()
```

Comment:
At first, the `simulated` multi-dc should work with the reservation.
At second, I believe it must be done on the config initialization step by checking that `use_capacity_reservation` is `False` when real multi-dc is enabled.

Reply:
the best would be if multi-dc scenarios were supported. 
This PR was aiming rather failure that happens when someone tries to use cr with multi-dc - to avoid it and show clearly that is not supported.
I think we could raise separate issue for that support and plan it if needed.

Reply:
I added verification if is simulated - in that case it's treated as single dc.

Reply:
https://github.com/scylladb/scylla-cluster-tests/issues/9317
fallback to regions is already done, CR's are tried over all regions in case of capacity issues.